packages/web-app-files/src/helpers/resource/move.ts

```diff
@@ -0,0 +1,166 @@
+      }
+    }
+    createModal(modal)
+  })
+}
+const resolveAllConflicts = async (
+  resourcesToMove,
+  targetFolder,
+  client,
+  createModal,
+  hideModal
+) => {
+  const targetFolderItems = await client.files.list(targetFolder.webDavPath, 50)
+  const targetPath = targetFolder.path
+  const index = targetFolder.webDavPath.lastIndexOf(targetPath)
+  const webDavPrefix = targetFolder.webDavPath.substring(0, index)
+
+  // Collect all conflicting resources
+  const allConflicts = []
+  for (const resource of resourcesToMove) {
+    const potentialTargetWebDavPath = join(webDavPrefix, targetFolder.path, resource.path)
+    const exists = targetFolderItems.some((e) => e.name === potentialTargetWebDavPath)
+    if (exists) {
+      allConflicts.push({
+        resource,
+        strategy: null
+      } as FileConflict)
+    }
+  }
+  let count = 0
+  let doForAllConflicts = false
+  let doForAllConflictsStrategy = null
+  const resolvedConflicts = []
+  for (const conflict of allConflicts) {
+    // Resolve conflicts accordingly
+    if (doForAllConflicts) {
+      conflict.strategy = doForAllConflictsStrategy
+      resolvedConflicts.push(conflict)
+      continue
+    }
+
+    // Resolve next conflict
+    const conflictsLeft = allConflicts.length - count
+    const result: ResolveConflict = await resolveFileExists(
+      createModal,
+      hideModal,
+      conflict.resource.name,
+      conflictsLeft
+    )
+    conflict.strategy = result.strategy
+    resolvedConflicts.push(conflict)
+    count += 1
+
+    // User checked 'do for all x conflicts'
+    if (!result.doForAllConflicts) continue
+    doForAllConflicts = true
+    doForAllConflictsStrategy = result.strategy
+  }
+  return resolvedConflicts
+}
+const showResultMessage = async(errors, movedResources, showMessage) => {
+  if (errors.length === 0) {
+    const count = movedResources.length
+    const title = `${count} item was moved successfully`
+    showMessage({
+      title,
+      status: 'success'
+    })
+    return 
+  }
+
+  if (errors.length === 1) {
+    const title = `Failed to move "${errors[0]?.resourceName}`
+    showMessage({
+      title,
+      status: 'danger'
+    })
+    return
+  }
+
+  const title = `Failed to move ${errors.length} resources`
+  showMessage({
+    title,
+    status: 'danger'
+  })
+}
+export const move = async (
+  resourcesToMove,
+  targetFolder,
+  client,
+  createModal,
+  hideModal,
+  showMessage
+) => {
+  const errors = []
+  const resolvedConflicts = await resolveAllConflicts(
+    resourcesToMove,
+    targetFolder,
+    client,
+    createModal,
+    hideModal
+  )
+  const movedResources = []
+
+  for (const resource of resourcesToMove) {
+    const hasConflict = resolvedConflicts.some((e) => e.resource.id === resource.id)
+    if (hasConflict) {
+      const resolveStrategy = resolvedConflicts.find((e) => e.resource.id === resource.id)?.strategy
+      if (resolveStrategy === ResolveStrategy.CANCEL) {
+        continue
```

Comment:
`cancel` sounds like it should abort the whole move process, not only the current file, i.e. I'd expect a `return` statement here.
`continue` would be the case for `skip`, which might be an additional resolve strategy actually. What do you think? 🤔

Reply:
Cancel equals skip, i think i messed up the wording ^^, but yes cancel implies to stop all